pkg/csi_driver/controller_test.go

```diff
@@ -1150,6 +1381,87 @@ func TestCreateSnapshot(t *testing.T) {
 	}
 }
 
+func TestCreateBackupURI(t *testing.T) {
+	backupName := "mybackup"
+	project := "test-project"
+	region := "us-central1"
+	instanceName := "myinstance"
+	shareName := "myshare"
+	cases := []struct {
+		name           string
+		backupName     string
+		backupLocation string
+		instance       *file.ServiceInstance
+		expectedURL    string
+		expectedRegion string
+		expectErr      bool
+	}{
+		{
+			name:           "Region is not provided. ServiceInstance is regional.",
```

Comment:
Some negative test cases as well needed here

Reply:
Good call. 

Actually, thinking about it, I have a question. Are we worried about cases where the provided `backupLocation` param is not a valid region? Right now if one is provided we just take it blindly as `region`. 

And if we are worried about that case, and a zone is provided instead of region there, do we want to crash out or take the region from the zone.

Reply:
Since this is a user provided field, my instinct is to error here, instead of trying to be cute, so I'll make that change. Open to other opinions though

Reply:
Pushed.

Reply:
If the location is unexpected, error is the right behavior